block_file.go

```diff
@@ -0,0 +1,131 @@
+package arcticdb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"hash/crc32"
+	"os"
+	"sync/atomic"
+)
+
+type BlockFile struct {
```

Comment:
Each page should have it's own checksum already in the Parquet file format right? The timestamp problem is interesting...

Reply:
I'm not sure if there's a good way for us to perform this in append mode while keeping the parquet file format which allows us to use the tool/oss ecosystem to inspect etc. I was looking at writing the table block as an appended row group in the file but I think there's some serious drawbacks to that approach. I think we'll want to switch gears here to writing the table block as a single full parquet file.

Reply:
Okay, we could switch again to the single block per file layout, even Create()/Open() a file is bit expensive

Reply:
In the Parquet file format the file metadata is contained in the footer, we'd have to append row groups until we closed out the file with the footer. Which means in the event of a crash we'd either have to throw all the row groups away, or we'd have to do some sort of file repair, and we couldn't easily read back the row groups without the file metadata. So for the sake of complexity (with the potential tradeoff on performance as you've rightly mentioned) I think we should start with dropping full Parquet files. We can always look into merging the files out of band later.

Reply:
Yea we'll need to have some compaction settings we can tweak but I think that compaction should be outside of the scope of this work to keep the complexity low.

We can always play around with memory mapping the files as well.

If we give the parquet filenames with a ULID we should be able use that to perform timestamp comparison no?

Reply:
> If we give the parquet filenames with a ULID we should be able use that to perform timestamp comparison no?

If we get rid of compaction for now, then yes, but we will have to handle this when dealing with compacted files, because they contain multiple blocks. For now, I think it is enought to use "collectionName:timestamp" as a filename.